addons/auth_passkeys/models/users_passkey.py

```diff
@@ -0,0 +1,90 @@
+    invalidation_counter = fields.Integer(default=0)
+
+    @property
+    def SELF_WRITEABLE_FIELDS(self):
+        return super().SELF_WRITEABLE_FIELDS + ['auth_passkeys_key_ids']
+
+    @property
+    def SELF_READABLE_FIELDS(self):
+        return super().SELF_READABLE_FIELDS + ['auth_passkeys_key_ids']
+    
+    def _get_session_token_fields(self):
+        return super()._get_session_token_fields() | {'invalidation_counter'}
+
+    def _check_credentials(self, password, env):
+        if self.env['res.users.apikeys']._check_credentials(scope='rpc', key=password) == self.env.uid:
+            return
+        if self.env.user.password_login_disabled:
+            raise AccessDenied()
+        return super()._check_credentials(password, env)
+    
+    @check_identity
+    def action_create_passkey(self):
+        return {
+            'name': _('Create Passkey'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'auth.passkeys.key',
+            'view_id': self.env.ref('auth_passkeys.auth_passkeys_key_create').id,
+            'view_mode': 'form',
+            'target': 'new',
+            'context': {
+                'dialog_size': 'medium',
+            }
+        }
+    
+    @check_identity
+    def action_delete_password_self(self):
+        return {
+            'name': _('Delete Password'),
+            'type': 'ir.actions.act_window',
+            'target': 'new',
+            'res_model': 'res.users',
+            'res_id': self.id,
+            'view_mode': 'form',
+            'view_id': self.env.ref('auth_passkeys.res_users_delete_password_self').id,
+        }
+    
+    @check_identity
+    def action_delete_password_other(self):
+        return {
+            'name': _('Delete Password'),
+            'type': 'ir.actions.act_window',
+            'target': 'new',
+            'res_model': 'res.users',
+            'res_id': self.id,
+            'view_mode': 'form',
+            'view_id': self.env.ref('auth_passkeys.res_users_delete_password_other').id,
+        }
+    
+    @check_identity
+    def delete_password(self):
+        for user in self:
+            if user.id != self.env.user.id:
+                user.password = '' # check access rights
+            self.env.cr.execute("UPDATE res_users SET password='' WHERE id=%s", (user.id,))
+    
+    def _compute_password_login_disabled(self):
+        # Password is disabled when set to ''
+        for user in self:
+            self.env.cr.execute(
+                "SELECT COALESCE(password, '') FROM res_users WHERE id=%s",
+                [user.id]
+            )
+            [hashed] = self.env.cr.fetchone()
+            valid, replacement = user._crypt_context()\
+                .verify_and_update('', hashed)
+            if replacement is not None:
+                self._set_encrypted_password(user.id, replacement)
+            user.password_login_disabled = valid
```

Comment:
This whole password deletion might be overkill in my opnion. What did Olivier say about this ?

Github doesn't provide this. Does someone else does it ? Who ?